### DISCUSSION TITLE
Better support for contact sensor

### DISCUSSION THREAD

**Discussion by Mis1ify** [make-all/tuya-local#634] [Q&A] [answered]
Hi, I managed to get my contact sensor working through YAML editing, but in comparation to cloud service, integration is really slow. It's because the contact sensor isn't always connected to the Wi-Fi, it connects after opening/closing door, which is causing problems like sensor is unavailable or is showing wrong information. Any ideas? It's my first custom added device, maybe it's my fault with the code, so here's my code:

> name: MENICI-WIFI-V01
primary_entity:
  entity: binary_sensor
  dps:
    - id: 1
      name: sensor
      type: boolean

**Comment by make-all** [comment, marked as answer]
It isn't possible to fix the fact that the device stays offline most of the time to conserve battery.  Such devices are always going to work better via the cloud.
